### DISCUSSION TITLE
Outsource the translations

### DISCUSSION THREAD

**Discussion by ariaieboy** [filamentphp/filament#5931] [Ideas]
Right now filament provides translations for its packages in several languages but it has some downsides to managing translations in the same repository.
first, everyone must download and ship all languages to the production even if they don't need translations.
Another problem is managing the missing translations in different languages.
right now the table translation file contains 196 lines of code in English 186 in fa 178 in bn etc.
that means each time a new key introduces translators must check the main file and find the missing keys then add the missing keys and then translate these keys.

we must either run some scripts to keep the keys synced between the locales or outsource the translations completely to another organization or another separate package.

I don't know if https://github.com/Laravel-Lang will accept third-party packages or not but if they do accept to manage the translations they can translate the locales to 78 languages with the help of the open source contributed and if needed with the use of bots.

**Comment by danharrin** [comment]
Thanks, but I don't agree with your arguments

1) There is essentially no overhead when installing all languages at once. Nothing gets shipped to users if your application only uses 1 locale.
2) We have [a command that you can use to check missing translations](https://github.com/filamentphp/filament#checking-for-missing-translations). There is no manually checking files.
3) I want the translations to remain under our control, they are a core part of Filament.
4) You don't have to use our translations, you can maintain your own in a separate plugin or package or within your app.
